### DISCUSSION TITLE
Custom telemetry output to graphical display

### DISCUSSION THREAD

**Discussion by flyingbandit** [iNavFlight/inav#6611] [Q&A] [answered]
Scale FPV Dash 

Suggested hardware with the graphics controller
https://4dsystems.com.au/uoled-160-g2?fbclid=IwAR0tULH-Dwmlp81Rl--6h9WARmJaffDMfbBUdN3E4ylO8PToSjj3PNoZq0s

Would it be possible to output the calculated horizon & home position/radar info over MSP/ uart output? 

I am willing to purchase & design / make a library of the screen end. I think it is possible that a custom telemetry output could be implemented into INAV to have direct communication between the screen & flight controller.

Users will be able to purchase the screen direct from 4dsystems & load firmware/ their own designs on there. As a semi open-source project if that makes sense.

If it is possible that a custom telemetry output can be implemented with horizon/ home/ radar, that would be fantastic.

This is a perfect solution for scale FPV cockpits!

**Comment by DzikuVx** [comment, marked as answer]
I think all the data you need are already part of MSP protocol and as such available over serial port. Also most of the data (maybe even all) are part of SmartPort telemetry

**Reply by flyingbandit** [reply]
Thank you. I will read the MSP library
